### PR TITLE
impl(otel): GCM exporter custom metric prefix

### PR DESCRIPTION
**Pull request by dbolduc**
Part of the work for #14074

- - -
This change is [Reviewable](https://reviewable.io/reviews/googleapis/google-cloud-cpp/14113)

### CODE REVIEW

File: google/cloud/opentelemetry/monitoring_exporter.h [hunk 1]
```diff
@@ -27,10 +27,35 @@ namespace cloud {
 namespace otel_internal {
 GOOGLE_CLOUD_CPP_INLINE_NAMESPACE_BEGIN
 
+/**
+ * The prefix prepended to metric names.
+ *
+ * The default value is `"workload.googleapis.com/"`. Note the trailing slash.
```
Comment (coryan) [inline google/cloud/opentelemetry/monitoring_exporter.h:33]: The quotes here are confusing.  It could be read as:

```cc
Options{}.set<MetricPrefixOption>("\"foo\"")
```

But you just mean:

```cc
Options{}.set<MetricPrefixOption>("foo")
```

File: google/cloud/opentelemetry/monitoring_exporter.cc [hunk 2]
```diff
@@ -57,17 +67,14 @@ class MonitoringExporter final
...
-    auto request = ToRequest(data);
+    auto request = ToRequest(data, prefix_);
     request.set_name(project_.FullName());
     if (request.time_series().empty()) {
       GCP_LOG(WARNING) << "Cloud Monitoring Export skipped. No TimeSeries "
                           "added to request.";
       return opentelemetry::sdk::common::ExportResult::kSuccess;
     }
 
-    // TODO(#13869) - Exporters of Google-defined metrics will need to use
```
Comment (coryan) [inline google/cloud/opentelemetry/monitoring_exporter.cc:68]: nit: unrelated?

**Review by coryan — APPROVED** [review]

File: google/cloud/opentelemetry/monitoring_exporter.cc (hunk 2 above)
Comment (dbolduc) [reply google/cloud/opentelemetry/monitoring_exporter.cc:68]: Correct, I got my commits twisted. Restored and updated the linked issue.

File: google/cloud/opentelemetry/monitoring_exporter.h (hunk 1 above)
Comment (dbolduc) [reply google/cloud/opentelemetry/monitoring_exporter.h]: Removed the backticks. Hopefully that is more clear.
